src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -1498,22 +2026,663 @@ static class D
+
+                    [InterceptsLocation("Program.cs", 22, 13)]
+                    public static bool NotReferenceEquals(object? obj1, object? obj2) => throw null!;
+                }
+            
+                public class Void { }
+                public struct Boolean { }
+                public class String { }
+                public class Attribute { }
+                public abstract class Enum { }
+                public enum AttributeTargets { }
+                public class AttributeUsageAttribute : Attribute
+                {
+                    public AttributeUsageAttribute(AttributeTargets targets) { }
+                    public bool AllowMultiple { get; set; }
+                    public bool Inherited { get; set; }
+                }
+                public class Exception { }
+                public abstract class ValueType { }
+                public struct Int32 { }
+                public struct Byte { }
+            }
+
+            namespace System.Runtime.CompilerServices
+            {
+                public sealed class InterceptableAttribute : Attribute { }
+
+                public sealed class InterceptsLocationAttribute : Attribute
+                {
+                    public InterceptsLocationAttribute(string filePath, int line, int column)
+                    {
+                    }
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(CreateEmptyCompilation((source, "Program.cs"), options: WithNullableEnable()), verify: Verification.Skipped);
+        verifier.Diagnostics.Where(d => d.Severity == DiagnosticSeverity.Error).Verify();
+
+        var referenceEqualsCallIL = """
+            {
+              // Code size        7 (0x7)
+              .maxstack  2
+              IL_0000:  ldarg.0
+              IL_0001:  ldarg.1
+              IL_0002:  bne.un.s   IL_0006
+              IL_0004:  ldnull
+              IL_0005:  throw
+              IL_0006:  ret
+            }
+            """;
+        verifier.VerifyIL("D.M0", referenceEqualsCallIL);
+        verifier.VerifyIL("D.M1", referenceEqualsCallIL);
+
+        verifier.VerifyIL("D.M2", """
+            {
+              // Code size       12 (0xc)
+              .maxstack  2
+              IL_0000:  ldarg.0
+              IL_0001:  ldarg.1
+              IL_0002:  call       "bool object.NotReferenceEquals(object, object)"
+              IL_0007:  brfalse.s  IL_000b
+              IL_0009:  ldnull
+              IL_000a:  throw
+              IL_000b:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void ReferenceEquals_02()
+    {
+        // Intercept a call to object.ReferenceEquals
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            static class D
+            {
+                public static void M0(object? obj1, object? obj2)
+                {
+                    if (object.ReferenceEquals(obj1, obj2))
```

Comment:
>object.ReferenceEquals(obj1, obj2) [](http://example.com/codeflow?start=24&length=34)

Should this be `obj1 == obj2`?

Reply:
No. An operator expression can't be intercepted.